tests/test_aea.py

```diff
@@ -112,6 +115,27 @@ def test_start_stop():
         agent.stop()
 
 
+def test_double_start():
+    """Tests the act function of the AEA."""
+    agent_name = "MyAgent"
+    private_key_path = os.path.join(CUR_PATH, "data", DEFAULT_PRIVATE_KEY_FILE)
+    builder = AEABuilder()
+    builder.set_name(agent_name)
+    builder.add_private_key(DEFAULT_LEDGER, private_key_path)
+    builder.add_skill(Path(CUR_PATH, "data", "dummy_skill"))
+    agent = builder.build()
+
+    with run_in_thread(agent.start, timeout=20):
+        wait_for_condition(
+            lambda: agent._main_loop and agent._main_loop.is_running, timeout=10
+        )
+        t = Thread(target=agent.start)
+        t.start()
+        sleep(1)
+        assert not t.is_alive()
```

Comment:
should it not be `assert is_alive()`?

Reply:
it's correct, agent should not start the second time and thread should exit instead of keep running